### DISCUSSION TITLE
Client using protocol revision 4 and 5

### DISCUSSION THREAD

**Discussion by schnabel** [miguelgrinberg/python-socketio#909] [Q&A] [answered]
Hi, I have to come up with a solution for a client that has to connect to servers with protocol revision 4 and to other servers using protocol revision 5. I guess tying the protocol revision to package versions has been a design decision with good reasons. The only idea I can think of is repackaging the latest 4.x release with a different name in case @miguelgrinberg does not mind.

**Comment by miguelgrinberg** [comment, marked as answer]
I would prefer that you don't release any packages with the same name as mine, as that is going to confuse people. A better option is for you to vendor the package versions that you need. In case you aren't familiar with this practice, the idea is to copy the code of the package into your own application, where you have full control over the naming.

**Reply by schnabel** [reply]
Thanks you for the quick answer. I will try to do that :-). May I ask why you did not create separate packages for the protocol revisions?

**Reply by miguelgrinberg** [reply]
That would be a very unusual thing to do. I don't see a point in supporting old revisions of the protocol.
